File: src/main/java/com/pubnub/api/models/server/FetchMessagesEnvelope.java
package com.pubnub.api.models.server;

import com.pubnub.api.models.consumer.history.PNFetchMessageItem;
import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
public class FetchMessagesEnvelope {
    private Map<String, List<PNFetchMessageItem>> channels;
    private FetchMessagesPage more;

    @Data
    public static class FetchMessagesPage {
        private Long start;
        private Long end;
        private Integer max;
    }
}

File: src/main/kotlin/com/pubnub/api/models/server/FetchMessagesEnvelope.kt
package com.pubnub.api.models.server

import com.pubnub.api.models.consumer.history.PNFetchMessageItem

data class FetchMessagesEnvelope(
    val channels: Map<String, List<PNFetchMessageItem>>,
    val more: FetchMessagesPage?
)

data class FetchMessagesPage(
    val start: Long? = null,
    val end: Long? = null,
    val max: Int? = null
)
